### PR TITLE
support dump fail log without coupling to travis environment

### PR DESCRIPTION
**Pull request by BusyJay**
Currently fail logs are only be printed in Travis CI. But we may need this too in general development or other CI environment, like reproducing a rare bug. So this pr add a target trace_test to meet this need.

@siddontang @disksing @iamxy PTAL

### CODE REVIEW

**Comment by BusyJay** [comment]
Debug locally or our internal CI.

**Comment by siddontang** [comment]
LGTM

PTAL @andelf

File: travis-build/Makefile [hunk 1]
```diff
@@ -67,7 +69,7 @@ prepare_brew:
 prepare_osx: ${LOCAL_DIR}/lib/librocksdb.dylib ${LOCAL_DIR}/bin/pd-server prepare-rustfmt | prepare_brew prepare_rust
 
 test_linux test_osx:
-	./travis-build/test.sh
+	export CI=true && ${CI_BUILD_DIR}/test.sh
```
Comment (andelf) [inline travis-build/Makefile]: CI=true ${CI_BUILD_DIR}/test.sh

when this variable is not used by other scripts.

File: travis-build/test.sh [hunk 2]
```diff
@@ -64,7 +67,7 @@ for l in sys.stdin:
     m = p.search(l)
     if m:
         cases.add(m.group(1).split(':')[-1])
-print '\n'.join(cases)
+print('\n'.join(cases))
```
Comment (andelf) [inline travis-build/test.sh]: print used as function, should better add:

``from __future__ import print_function``

File: travis-build/Makefile (hunk 1 above)
Comment (BusyJay) [inline travis-build/Makefile]: I don't get it, can you explain it in details?

**Comment by BusyJay** [comment]
@andelf PTAL

**Comment by andelf** [comment]
LGTM

**Pull request merged**